### ISSUE TITLE
No minimize and maximize buttons on a window bar on the latest pyface and linux

### ISSUE THREAD

**Issue by bebaek** [enthought/pyface#802]
This happens only for my ubuntu 20.04 laptop. Windows VM is fine. Also internal custom app is fine.

The problematic environment has: CPython 3.6.12+2, pyface 7.1.0-1, traitsui 7.1.0-1. Another environment with older runtime/packages seem fine though I didn't test extensively.

Test code:

```python
from traits.api import *
from traitsui.api import *

class Foo(HasTraits):
    bar = Str()

    def default_traits_view(self):
        return View(
            Item('bar'),
            width=400,
            height=400,
            resizable=True,
        )

Foo().configure_traits()
```

![Screenshot from 2020-11-12 15-34-27](https://user-images.githubusercontent.com/27637142/98999694-d1326a00-24fd-11eb-9618-1522e92df4a1.png)

**Comment by corranwebster** [comment]
This is indeed odd.  Which Qt version are you using?

**Comment by kitchoi** [comment]
I could reproduce this on a Linux machine, and after falling back to traitsui 7.0.1 and pyface 7.0.1, the behaviour still persists, so at least this is not a regression from Pyface/TraitsUI 7.0.1 to 7.1.0

**Comment by kitchoi** [comment]
The dependencies I used:
```
Package    Version
---------- -------
pip        19.2.3 
pyface     7.0.1  
Pygments   2.7.2  
PySide2    5.15.1 
setuptools 41.2.0 
shiboken2  5.15.1 
traits     6.1.1  
traitsui   7.0.1
```
This is on Python 3.8 and Linux.

**Comment by bebaek** [comment]
> This is indeed odd. Which Qt version are you using?

qt 5.12.6-6

I see things are fine in our previous dev env with: cpython 3.6.0+12, qt 5.12.6-5, pyface 7.0.1-2, traitsui 7.0.1-2

**Comment by bebaek** [comment]
> I could reproduce this on a Linux machine, and after falling back to traitsui 7.0.1 and pyface 7.0.1, the behaviour still persists, so at least this is not a regression from Pyface/TraitsUI 7.0.1 to 7.1.0

Somehow, that downgrade to 7.0.1 worked for me.

**Comment by kitchoi** [comment]
Thanks @bebaek 

This is really bizarre. Now I try a Ubuntu 16 (Xenial) VM, and I could not reproduce:
```
$ edm install traitsui pyqt5
$ python scripts/pyface-802.py
```
(The script contains the code snippet from this issue)

And I see this:
![Screenshot from 2020-11-18 13-02-24](https://user-images.githubusercontent.com/3673984/99533794-5dc9a600-299e-11eb-9671-b04ba45f4e76.png)

I wonder if there are other parameters we are overlooking?

Packages:
```
appdirs            1.4.3-1   enthought/free
distribute_remove  1.0.0-4   enthought/free
freetype           2.10.0-6  enthought/free
harfbuzz           2.6.4-4   enthought/free
icu4c              50.2-2    enthought/free
libjpeg            7.0-5     enthought/free
libpng             1.6.36-5  enthought/free
libxkbcommon       0.9.1-3   enthought/free
libxml2            2.9.7-1   enthought/free
packaging          16.8-4    enthought/free
pip                20.0.2-2  enthought/free
pyface             7.1.0-1   enthought/free
pygments           2.2.0-1   enthought/free
pyparsing          2.2.0-1   enthought/free
pyqt5              5.14.2-3  enthought/free
pyqt5_sip          12.7.2-1  enthought/free
qt                 5.12.6-6  enthought/free
setuptools         41.6.0-4  enthought/free
six                1.15.0-1  enthought/free
traits             6.1.1-1   enthought/free
traitsui           7.1.0-1   enthought/free
```

I have also exported the EDM environment I used here: 
<details>
    <summary> EDM bundle </summary>

```
{
    "mark": [
        {
            "name": "appdirs",
            "state": "auto"
        },
        {
            "name": "distribute_remove",
            "state": "auto"
        },
        {
            "name": "freetype",
            "state": "auto"
        },
        {
            "name": "harfbuzz",
            "state": "auto"
        },
        {
            "name": "icu4c",
            "state": "auto"
        },
        {
            "name": "libjpeg",
            "state": "auto"
        },
        {
            "name": "libpng",
            "state": "auto"
        },
        {
            "name": "libxkbcommon",
            "state": "auto"
        },
        {
            "name": "libxml2",
            "state": "auto"
        },
        {
            "name": "packaging",
            "state": "auto"
        },
        {
            "name": "pip",
            "state": "manual"
        },
        {
            "name": "pyface",
            "state": "auto"
        },
        {
            "name": "pygments",
            "state": "auto"
        },
        {
            "name": "pyparsing",
            "state": "auto"
        },
        {
            "name": "pyqt5",
            "state": "manual"
        },
        {
            "name": "pyqt5_sip",
            "state": "auto"
        },
        {
            "name": "qt",
            "state": "auto"
        },
        {
            "name": "setuptools",
            "state": "manual"
        },
        {
            "name": "six",
            "state": "auto"
        },
        {
            "name": "traits",
            "state": "auto"
        },
        {
            "name": "traitsui",
            "state": "manual"
        }
    ],
    "metadata_version": "2.0",
    "modifiers": {
        "allow_any": [],
        "allow_newer": [],
        "allow_older": []
    },
    "repositories": {
        "packages": [
            "enthought/commercial",
            "enthought/free"
        ],
        "runtimes": [
            "enthought/free"
        ]
    },
    "requirements": [
        {
            "name": "appdirs",
            "repository": "enthought/free",
            "sha256": "25ac2fe7f2f810d6de4610728a7794a8f70439c3f8fb2af2e6920938e666f67d",
            "version": "1.4.3-1"
        },
        {
            "name": "distribute_remove",
            "repository": "enthought/free",
            "sha256": "01faf0d5a61536723ef6b570b30361056021203cff1f9882e8a842ad2ef20e3e",
            "version": "1.0.0-4"
        },
        {
            "name": "freetype",
            "repository": "enthought/free",
            "sha256": "71b73bc7c8d5dcbb7c76822fde27255ef72c97653df3f2bdd8b6ce73513088ec",
            "version": "2.10.0-6"
        },
        {
            "name": "harfbuzz",
            "repository": "enthought/free",
            "sha256": "9cad02b85538305109924c576edbd787d9d1943e95229b17760e9a702322357a",
            "version": "2.6.4-4"
        },
        {
            "name": "icu4c",
            "repository": "enthought/free",
            "sha256": "03d2dd4d57c99076ffefe2f767302301d6d2067d26caa530d680556ea30e4927",
            "version": "50.2-2"
        },
        {
            "name": "libjpeg",
            "repository": "enthought/free",
            "sha256": "1b053bd058418d097fec5e0ec7a90d28ac5a0a5b08611ca401a990078d84b8a3",
            "version": "7.0-5"
        },
        {
            "name": "libpng",
            "repository": "enthought/free",
            "sha256": "977270556073a5bed2e2dd8c842c6d76d952c83d0b67461e89b6f67652003651",
            "version": "1.6.36-5"
        },
        {
            "name": "libxkbcommon",
            "repository": "enthought/free",
            "sha256": "c750e59293530d7e14fbf0f91ec40df0ab765c9a9d4dfaae4ce07ebdb84aace9",
            "version": "0.9.1-3"
        },
        {
            "name": "libxml2",
            "repository": "enthought/free",
            "sha256": "0af8cbf11994b577685367ab633eb400494f04b268a51a010d9b087423d6412f",
            "version": "2.9.7-1"
        },
        {
            "name": "packaging",
            "repository": "enthought/free",
            "sha256": "4c13be5bf253485602b891a11320eaaec7bb847996c86b27bee0f93a46f6ca28",
            "version": "16.8-4"
        },
        {
            "name": "pip",
            "repository": "enthought/free",
            "sha256": "66e65ff3d3ca34c809c5a2cb6bb118da2aea13e76dcc8a1d860e3170360a8d71",
            "version": "20.0.2-2"
        },
        {
            "name": "pyface",
            "repository": "enthought/free",
            "sha256": "b4f44ac63effa34528509187d72cc1105ca194e80bb338af70e825e8809e9999",
            "version": "7.1.0-1"
        },
        {
            "name": "pygments",
            "repository": "enthought/free",
            "sha256": "7e7d360cc29444090f3fc9e621eb073a3ed084aefcbad7adf862b29498624931",
            "version": "2.2.0-1"
        },
        {
            "name": "pyparsing",
            "repository": "enthought/free",
            "sha256": "84352e1c11b18b08eb059b5ef008356e49601fceba7119389e131a694ac046ba",
            "version": "2.2.0-1"
        },
        {
            "name": "pyqt5",
            "repository": "enthought/free",
            "sha256": "7c0e3d49b3809ba1c193238e4ee73660755ddf4fda1ef23c3d82372fe93f7671",
            "version": "5.14.2-3"
        },
        {
            "name": "pyqt5_sip",
            "repository": "enthought/free",
            "sha256": "40efb786282f22e6e43b02e85a82a8c47b804d703ebcb0d9d457ad9c5d907501",
            "version": "12.7.2-1"
        },
        {
            "name": "qt",
            "repository": "enthought/free",
            "sha256": "95dedf063b86ac78cf9c085c72480718d528c6087b650dfa0ec4b3d9adf9900d",
            "version": "5.12.6-6"
        },
        {
            "name": "setuptools",
            "repository": "enthought/free",
            "sha256": "9a1c3b5592dfe6f502d818ee0a3ad81e138a9c02d88579966780655ebcf3d553",
            "version": "41.6.0-4"
        },
        {
            "name": "six",
            "repository": "enthought/free",
            "sha256": "43c02e18c6d64b2746caa217283a9ef42e61dc9921da2e3aec77f57fafbd2c35",
            "version": "1.15.0-1"
        },
        {
            "name": "traits",
            "repository": "enthought/free",
            "sha256": "a3e49aaba4e46da90bcad9b09e94f2a6102e6a82c8bc468dba325e0053e779e1",
            "version": "6.1.1-1"
        },
        {
            "name": "traitsui",
            "repository": "enthought/free",
            "sha256": "ed0a8fd8f866849976cfd4d1e6547bed65a63d4234e24030b65232df39a5d590",
            "version": "7.1.0-1"
        }
    ],
    "runtime": {
        "implementation": "cpython",
        "platform": "rh7_x86_64",
        "platform_abi": "gnu",
        "repository": "enthought/free",
        "sha256": "e05dda78ee88d9cd30d9d6feba04df8cbc1b954f934e6d6a7232b73b64f9668c",
        "version": "3.6.12+2"
    }
}
```

</details>

**Comment by kitchoi** [comment]
Ah the Ubuntu version I used is 18, but the issue is observed on Ubuntu 20. I will try again.

**Comment by kitchoi** [comment]
Upon trying a Ubuntu 18 VM, I am now able to reproduce.

**Comment by kitchoi** [comment]
I think I could narrow this down to this change in https://github.com/enthought/traitsui/pull/1083

Unfortunately, reverting it will change behaviour for CentOS Linux in another undesirable way, reopening https://github.com/enthought/traitsui/issues/1078
Essentially, we are facing this X11 Peculiarities noted in Qt documentation: https://doc.qt.io/qt-5/application-windows.html#x11-peculiarities

> X11 Peculiarities
On X11, a window does not have a frame until the window manager decorates it. This happens asynchronously at some point in time after calling QWidget::show() and the first paint event the window receives, or it does not happen at all. Bear in mind that X11 is policy-free (others call it flexible). Thus you cannot make any safe assumption about the decoration frame your window will get. **Basic rule: There's always one user who uses a window manager that breaks your assumption, and who will complain to you.**

> Furthermore, a toolkit cannot simply place windows on the screen. All Qt can do is to send certain hints to the window manager. The window manager, a separate process, may either obey, ignore or misunderstand them. Due to the partially unclear Inter-Client Communication Conventions Manual (ICCCM), window placement is handled quite differently in existing window managers.

**Comment by kitchoi** [comment]
On the same Ubuntu VM where I could reproduce the issue, I could switch the desktop environment from Ubuntu to LXDE:
![Screenshot 2020-11-18 at 16 38 43](https://user-images.githubusercontent.com/3673984/99559470-bb6ceb00-29bc-11eb-9999-9106327e65c9.png)

And there the `WindowMinimizeButtonHint` and `WindowMaximizeButtonHint` are respected such that the maximize and minimize buttons are shown (see [code](https://github.com/enthought/traitsui/blob/770e94346e06dc70909f125ba7237edce90872da/traitsui/qt4/ui_base.py#L139-L140)), and the window can be maximized to fill up the screen.
![Screenshot 2020-11-18 at 16 39 40](https://user-images.githubusercontent.com/3673984/99559779-0c7cdf00-29bd-11eb-9fa0-ce5a85711689.png)

**Comment by kitchoi** [comment]
In https://github.com/enthought/pyface/issues/802#issuecomment-726720356

> I could reproduce this on a Linux machine, and after falling back to traitsui 7.0.1 and pyface 7.0.1, the behaviour still persists, so at least this is not a regression from Pyface/TraitsUI 7.0.1 to 7.1.0

The Linux machine I tried then was a CentOS... whose window manager does not actually do minimize or maximize buttons, I jumped into conclusions. In that particular window manager, the maximize action is achieved by double clicking the window frame. And I can confirm that on Pyface/TraitsUI 7.0.1, such a window can be maximized by double clicking on CentOS, but not any more with Pyface/TraitsUI 7.1.0.
